pkg/dao/repository.go

```diff
@@ -0,0 +1,42 @@
+package dao
+
+import (
+	"fmt"
+
+	"github.com/content-services/content-sources-backend/pkg/api"
+	"github.com/content-services/content-sources-backend/pkg/db"
+	"github.com/content-services/content-sources-backend/pkg/models"
+)
+
+type repositoryDaoImpl struct {
+}
+
+func GetRepositoryDao() repositoryDaoImpl {
+	return repositoryDaoImpl{}
+}
+
+func (r repositoryDaoImpl) Create(newRepo api.Repository) error {
+
+	newRepoConfig := models.RepositoryConfiguration{
+		Name:      newRepo.Name,
+		URL:       newRepo.URL,
+		Version:   newRepo.DistributionVersion,
+		Arch:      newRepo.DistributionArch,
+		AccountID: newRepo.AccountID,
+		OrgID:     newRepo.OrgID,
+	}
+
+	dest := models.RepositoryConfiguration{}
+	result := db.DB.Where("org_id = ? AND url = ?", newRepo.OrgID, newRepo.URL).Find(&dest)
+	if result.Error != nil {
+		return result.Error
+	}
+	if result.RowsAffected > 0 {
+		return fmt.Errorf("repository with this URL already belongs to organization")
+	}
```

Comment:
would it make sense to just create a uniquness constraint and check for that error on create?